### PR TITLE
docs: Updating documentation for bzlmod

### PR DESCRIPTION
**Pull request by chrislovecnm**
- Updated primary README.md to include documentation for using bzlmod or a WORKSPACE file.
- Updated gazelle/README.md to include documentation for only using bzlmod and provided a link to the older docs.
- Included other general updates for the gazelle documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +157,22 @@ target in the appropriate wheel repo.
...
+To add pip dependencies to your `MODULES.bazel` file, use the `pip.parse` extension, and call it to create the
+central external repo and individual wheel external repos.
+
+```python
+pip.parse(
+    name = "my_deps",
+    requirements_lock = "//:requirements_lock.txt",
+)
+
+use_repo(pip, "pip")
```
Comment (f0rmiga) [inline README.md:171]: Doesn't this have to match the name you gave above?

File: README.md [hunk 2]
```diff
@@ -33,6 +33,45 @@ contribute](CONTRIBUTING.md) page for information on our development workflow.
...
+#### Toolchain registration with bzlmod
+
+To register a hermetic Python toolchain rather than rely on a system-installed interpreter for runtime execution, you can add to the `MODULES.bazel` file:
+
+```python
+# Find the latest version number here: https://github.com/bazelbuild/rules_python/releases
+# and change the version number if needed in the line below.
+bazel_dep(name = "rules_python", version = "0.20.0")
+
+pip = use_extension("@rules_python//python:extensions.bzl", "pip")
```
Comment (rickeylev) [inline README.md:54]: nit: I don't think the pip part is necessary just to use the hermetic toolchains? Most people will want there anyways, though, so I don't think it's NBD to leave it here.

File: gazelle/README.md [hunk 3]
```diff
@@ -1,21 +1,48 @@
...
-It requires Go 1.16+ to compile.
+We have an example of using Gazelle with Python located [here](https://github.com/bazelbuild/rules_python/tree/main/examples/build_file_generation).
 
-## Installation
+## Adding Gazelle to your project
 
-First, you'll need to add Gazelle to your `WORKSPACE` file.
-Follow the instructions at https://github.com/bazelbuild/bazel-gazelle#running-gazelle-with-bazel
+First, you'll need to add Gazelle to your `MODULES.bazel` file.
+Get the current version of Gazelle from there releases here:  https://github.com/bazelbuild/bazel-gazelle/releases/
```
Comment (f0rmiga) [inline gazelle/README.md:23]:
```suggestion
Get the current version of Gazelle from there releases here: https://github.com/bazelbuild/bazel-gazelle/releases/.
```

File: gazelle/README.md [hunk 4]
```diff
@@ -1,21 +1,48 @@
...
-depends on.
 
-See the installation `WORKSPACE` snippet on the Releases page:
-https://github.com/bazelbuild/rules_python/releases
+See the installation `MODULE.bazel` snippet on the Releases page:
+https://github.com/bazelbuild/rules_python/releases in order to configure rules_python.
+
+You will also need to add the `bazel_dep` for configuration for `rules_python_gazelle_plugin`.
+
+Here is an example snippet of a `MODULE.bazel` file.
```
Comment (f0rmiga) [inline gazelle/README.md:31]:
```suggestion
Here is a snippet of a `MODULE.bazel` file.
```

File: gazelle/README.md [hunk 5]
```diff
@@ -1,21 +1,48 @@
...
+bazel_dep(name = "rules_python", version = "0.20.0")
+
+# The following stanza defines the dependency rules_python.
+# For typical setups you set the version.
+bazel_dep(name = "rules_python_gazelle_plugin", version = "0.20.0")
+
+# The following stanza defines the dependency rules_python.
+bazel_dep(name = "gazelle", version = "0.30.0", repo_name = "bazel_gazelle")
+```
+You will also need to do the other usual configuration for `rules_python`. In your
```
Comment (f0rmiga) [inline gazelle/README.md:44]:
```suggestion
You will also need to do the other usual configuration for `rules_python` in your
```

File: gazelle/README.md [hunk 6]
```diff
@@ -157,11 +184,29 @@ Next, all source files are collected into the `srcs` of the `py_library`.
 Finally, the `import` statements in the source files are parsed, and
 dependencies are added to the `deps` attribute.
 
-### Tests
+### Unit Tests
+
+A `py_test` target is added to the BUILD file when gazelle encounters
+a file named `__test__.py`.
+Often Python unit test files are named with the suffix `_test`.
+For example if we had a folder that is a package named "foo" we could have a Python file named `foo_test.py`
```
Comment (f0rmiga) [inline gazelle/README.md:192]:
```suggestion
For example, if we had a folder that is a package named "foo" we could have a Python file named `foo_test.py` 
```

File: gazelle/README.md [hunk 7]
```diff
@@ -157,11 +184,29 @@ Next, all source files are collected into the `srcs` of the `py_library`.
 Finally, the `import` statements in the source files are parsed, and
 dependencies are added to the `deps` attribute.
 
-### Tests
+### Unit Tests
+
+A `py_test` target is added to the BUILD file when gazelle encounters
+a file named `__test__.py`.
+Often Python unit test files are named with the suffix `_test`.
```
Comment (f0rmiga) [inline gazelle/README.md:191]:
```suggestion
Often, Python unit test files are named with the suffix `_test`.
```

File: gazelle/README.md [hunk 8]
```diff
@@ -157,11 +184,29 @@ Next, all source files are collected into the `srcs` of the `py_library`.
 Finally, the `import` statements in the source files are parsed, and
 dependencies are added to the `deps` attribute.
 
-### Tests
+### Unit Tests
```
Comment (f0rmiga) [inline gazelle/README.md:187]: Gazelle now supports generating 1 target per test it finds when `__test__.py` or `__test__` target doesn't exist.

File: gazelle/README.md [hunk 9]
```diff
@@ -170,16 +215,18 @@ of a Python program.
...
+Python interpreter as a subprocess to parse Python source files. 
+See the gazelle documentation https://github.com/bazelbuild/bazel-gazelle/blob/master/extend.md 
+for more information on extending Gazelle.
 
-The Go dependencies are managed by the go.mod file.
-After changing that file, run `go mod tidy` to get a `go.sum` file,
-then run `bazel run //:update_go_deps` to convert that to the `gazelle/deps.bzl` file.
-The latter is loaded in our `/WORKSPACE` to define the external repos
-that we can load Go dependencies from.
+If you add a new Go dependencies the plugin source code you need to "tidy" the go.mod file.
```
Comment (f0rmiga) [inline gazelle/README.md:225]:
```suggestion
If you add new Go dependencies to the plugin source code, you need to "tidy" the go.mod file. 
```

File: gazelle/README.md (hunk 8 above)
Comment (chrislovecnm) [reply gazelle/README.md:187]: How would your recommend saying that in more detail?

Comment (f0rmiga) [reply gazelle/README.md:187]: Something like:

```
Gazelle will detect test files that meet the `test_*.py` and `*_test.py` patterns and add one `py_test` per test detected. If you want to group multiple test files into a single `py_test`, use the `__test__.py` or `__test__`, bla bla bla bla bla
```

**Review by rickeylev — APPROVED** [review]

**Comment by rickeylev** [comment]
I'm going to merge this despite that comment being addressed because the main point of this PR is to address the bzlmod instructions.

**Pull request merged**